results/cmd/api/main_test.go

```diff
@@ -437,3 +439,146 @@ func TestDeleteTaskRun(t *testing.T) {
 		t.Fatalf("same taskrun not supposed to be deleted again: %v", r.String())
 	}
 }
+func TestListTaskRuns(t *testing.T) {
+	// Create a temporary database
+	srv, err := setupTestDB(t)
+	if err != nil {
+		t.Fatalf("failed to setup db: %v", err)
+	}
+
+	// Create a bunch of taskruns for testing
+	t1 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "31415926",
+			Name:      "taskrun",
+			Namespace: "default",
+		},
+	}
+	t2 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "43245243",
+			Name:      "task",
+			Namespace: "default",
+		},
+	}
+	t3 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "1234556",
+			Name:      "mytaskrun",
+			Namespace: "demo",
+		},
+	}
+	t4 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "demo",
+		},
+	}
+	t5 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "official",
+		},
+	}
+	testList := []*pb.TaskRun{t1, t2, t3, t4, t5}
+	tt := []struct {
+		name   string
+		list   []*pb.TaskRun
+		filter string
+		expect []*pb.TaskRun
+	}{
+		{
+			name:   "test simple query",
+			list:   testList,
+			filter: `taskrun.api_version=="v1beta1"`,
+			expect: []*pb.TaskRun{t1, t3, t4},
+		},
+		{
+			name:   "test logical &&",
+			list:   testList,
+			filter: `taskrun.api_version=="v1beta1" && taskrun.metadata.namespace=="demo"`,
+			expect: []*pb.TaskRun{t3, t4},
+		},
+		{
+			name:   "test logical ||",
+			list:   testList,
+			filter: `taskrun.metadata.name=="newtaskrun" || taskrun.metadata.namespace=="default"`,
+			expect: []*pb.TaskRun{t1, t2, t4, t5},
+		},
+		{
+			name:   "test startWith",
+			list:   testList,
+			filter: `taskrun.metadata.name.startsWith("new")`,
+			expect: []*pb.TaskRun{t4, t5},
+		},
+		{
+			name:   "test endWith",
+			list:   testList,
+			filter: `taskrun.metadata.name.endsWith("run")`,
+			expect: []*pb.TaskRun{t1, t3, t4, t5},
+		},
+		{
+			name:   "test empty filter",
+			list:   testList,
+			filter: "",
+			expect: testList,
+		},
+		{
+			name:   "test invalid field",
+			list:   testList,
+			filter: `task.name=="newtaskrun"`,
+		},
+		{
+			name:   "test invalid value type",
+			list:   testList,
+			filter: `taskrun.name==newtaskrun`,
+		},
+		{
+			name:   "test value not existing in the server record",
+			list:   testList,
+			filter: `taskrun.metadata.name=="notaskrun"`,
+			expect: nil,
+		},
```

Comment:
Other interesting tests that come to mind:
- What if the user enters a non-boolean (but valid) filter?
- What if a user enters in a random word instead of a boolean expression? (e.g. `tekton`)
- Does case sensitivity matter?

Reply:
done